src/main/java/moze_intel/projecte/handlers/InternalAbilities.java

```diff
@@ -76,23 +79,40 @@ public void tick()
 			gemChestCooldown--;
 		}
 
-		if (!shouldPlayerFly() && hadFlightItem)
+		if (!shouldPlayerFly())
 		{
-			if (player.capabilities.allowFlying)
+			if (hadFlightItem)
 			{
-				PlayerHelper.updateClientServerFlight(player, false);
+				if (player.capabilities.allowFlying)
+				{
+					PlayerHelper.updateClientServerFlight(player, false);
+				}
+
+				hadFlightItem = false;
 			}
-			
-			hadFlightItem = false;
+			wasFlyingGamemode = false;
+			wasFlying = false;
 		}
-		else if(shouldPlayerFly() && !hadFlightItem)
+		else //if (shouldPlayerFly())
```

Comment:
any reason why it's commented and not removed?

Reply:
Mainly just so it is clearer what that branch is at a glance. I can remove it if you wish.

Reply:
yeah, remove it - it's visible by just looking at the if condition

Reply:
Done.